tmt/result.py

```diff
@@ -363,6 +374,24 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
 
         return self
 
+    def to_subresult(self, invocation: 'tmt.steps.execute.TestInvocation') -> 'SubResult':
+        """ Convert result to tmt subresult """
+
+        # TODO: Try to find a better way for conversion (e.g. using
+        # from_spec/to_spec/from_serialized/to_serialized).
+        return SubResult(
+            # Add subresult name under the test namespace
+            name=f"{invocation.test.name}/{self.name.lstrip('/')}",
+            result=self.result,
+            note=self.note,
+            log=self.log,
+            start_time=self.start_time,
+            end_time=self.end_time,
+            duration=self.duration,
+
+            # Also convert all CheckResult instances to SubCheckResult instances
+            check=[c.to_subcheck() for c in self.check])
```

Comment:
A single-letter variable that's not `i` or `j` :) See https://github.com/teemtee/tmt/pull/3200/files#r1799649164

Reply:
:D, fixed.

What about adding some automatic lint checks for these single-letter variable names in ruff/pylint?
CC: @martinhoyer